src/app/store/controller/selectors.ts

```diff
@@ -373,6 +374,34 @@ const getByFabricId = createSelector(
       }, [])
 );
 
+/**
+ * Get all region controllers and separate by their vault configuration status.
+ * @param state - The redux state.
+ * @returns Two lists of region controllers - one where none are configured with vault, the other where all are configured with vault.
+ */
+const getVaultConfiguredControllers = createSelector(
+  [defaultSelectors.all],
+  (controllers: Controller[]) => {
+    const regionControllers = controllers.filter((controller) => {
```

Comment:
Ideally `regionControllers` should be its own selector which value would be cached.

Reply:
That's fine, implementing that